06-building-decays.md

```diff
@@ -0,0 +1,161 @@
+from PhysSelPython.Wrappers import Selection
+d0_sel = Selection("Sel_D0",
+                   Algorithm=d0,
+                   RequiredSelections=[Pions, Kaons])
+```
+
+Now we can use another `CombineParticles` to build the D* with pions and the D0's as inputs, and applying a filtering only on the soft pion:
+
+```python
+dstar = CombineParticles("CombineDstar",
+                         Decay='[D*(2010)+ -> D0 pi+]cc',
+                         DaughtersCuts={'pi+': '(TRCHI2DOF < 3) & (PT > 100*MeV)'}
+                         CombinationCut="(ADAMASS('D*(2010)+') < 400*MeV)",
+                         MotherCut="(abs(M-MAXTREE('D0'==ABSID,M)-145.42) < 10*MeV) & (VFASPF(VCHI2/VDOF)< 9)")
+dstar_sel = Selection("Sel_Dstar",
+                      Algorithm=dstar,
+                      RequiredSelections=[d0_sel, Pions])
+```
+
+> ## Performing shared selections {.callout}
+> In some cases we may want to build several decays in the same script with some common particles/selection;
+> for example, in our case we could have been building D0->KK in the same script, and then we would have wanted to select the soft pion in the same way when building the D*.
+> In this situation, we can make use of the `FilterDesktop` algorithm, which takes a TES location and filters the particles inside according to a given LoKi functor in the `Code` property, which then can be given as input to a `Selection`:
+> 
+> ```python
+> from Configurables import FilterDesktop
+> soft_pion = FilterDesktop("Filter_SoftPi",
+>                           Code='(TRCHI2DOF < 3) & (PT > 100*MeV)')
+> soft_pion_sel = Selection("Sel_SoftPi",
+>                           Algorithm=soft_pion,
+>                           RequiredSelections=[Pions])
+> dstar = CombineParticles("CombineDstar",
+>                          Decay='[D*(2010)+ -> D0 pi+]cc',
+>                          CombinationCut="(ADAMASS('D*(2010)+') < 400*MeV)",
+>                          MotherCut="(abs(M-MAXTREE('D0'==ABSID,M)-145.42) < 10*MeV) & (VFASPF(VCHI2/VDOF)< 9)")
+> dstar_sel = Selection("Sel_Dstar",
+>                       Algorithm=dstar,
+>                       RequiredSelections=[d0_sel, soft_pion_sel])
+> ```
+>
+> This allows to (a) save time by performing the filtering of the soft pions only once, and (b) keep all the common cuts in a single place, avoiding duplication of code.
+
+
+We can now build build a `SelectionSequence` to add to the `DaVinci` execution sequence
+
+```python
+from PhysSelPython.Wrappers import SelectionSequence
+dstar_seq = SelectionSequence('Dstar_Seq', TopSelection=dstar_sel)
+from Configurables import DaVinci
+DaVinci().UserAlgorithms += [dstar_seq.sequence()]
+```
+
+Now you can finish the script (the base of which can be found [here](code/06-building-decays/build_decays.py)) by adapting the basic `DaVinci` configuration from its corresponding [lesson](09-minimal-dv-job.html) and check the output ntuple.
+
+Congratulations! You've built the base of a Stripping line!
```

Comment:
“Stripping” → “stripping” I think?

IMO this sentence comes a bit out of the blue, because the relation of this to the stripping hasn't yet been discussed. You could modify the first paragraph to introduce the idea:

> In order to perform most physics analyses we need to build a decay chain with reconstructed (or MC) particles that represents the physics process we want to study. In LHCb, this decay chain can be built through `LHCb::Particle` (`LHCb::MCParticle`) objects that represent individual particles and contain links to their children, also represented by the same type of object.
> 
> For collision data, this particle combination is most often done in a central production called the _stripping_, where a stripping _line_ creates a specific decay chain and applies some selection criteria to it.